src/install.c

```diff
@@ -1144,7 +1144,6 @@ void install_args_free(RaucInstallArgs *args)
 {
 	g_free(args->name);
 	g_mutex_clear(&args->status_mutex);
-	g_assert_cmpint(args->status_result, >=, 0);
```

Comment:
Why did this fail only now? In src/main.c, we've been freeing install args for a long time.

Reply:
Ok, I had a closer look at this now as I knew I came across this but did not recap where actually.

The potential problem I referred to is that `install_args_new()` initializes `status_result` to `-2`.
Thus if `install_args_free()` gets called before some instance changed this value, we will end up in the assertion error. But maybe, yes, these probably are the exact cases that we wanted to find with the assertion...

The only two points I dug out where this could have been an issue are:

https://github.com/rauc/rauc/blob/master/src/service.c#L63

and

https://github.com/rauc/rauc/blob/master/src/install.c#L1159

As they potentially return without having the args touched.

BUT, as we do

https://github.com/rauc/rauc/blob/master/src/service.c#L80

(using only `g_free()` on the complex initialized type).This will never trigger.

IMHO this should call `install_args_free()` here.
Also setting `args = NULL` above seems to be a bit unsafe but should not make any issues if we properly cleared args before.

Reply:
This should be solved now.